packages/graphback-codegen-schema/tests/GraphQLSchemaCreator.test.ts

```diff
@@ -405,14 +405,58 @@ test('Transient field is excluded from input types', () => {
   const schema = schemaGenerator.transformSchema(metadata);
 
   const createUserInput = assertInputObjectType(schema.getType('CreateUserInput'))
-  expect(Object.keys(createUserInput.getFields())).toEqual(['id', 'name'])
+  expect(Object.keys(createUserInput.getFields()).includes('generatedFullName')).toEqual(false);
 
   const mutateUserInput = assertInputObjectType(schema.getType('MutateUserInput'))
   expect(Object.keys(mutateUserInput.getFields())).toEqual(['id', 'name'])
 
   const userFilterInput = assertInputObjectType(schema.getType('UserFilter'))
-  expect(Object.keys(userFilterInput.getFields())).toEqual(['id', 'name', 'and', 'or', 'not'])
+  expect(Object.keys(userFilterInput.getFields()).includes('generatedFullName')).toEqual(false);
 
   const userSubscriptionInput = assertInputObjectType(schema.getType('UserFilter'))
-  expect(Object.keys(userSubscriptionInput.getFields())).toEqual(['id', 'name', 'and', 'or', 'not'])
+  expect(Object.keys(userSubscriptionInput.getFields()).includes('generatedFullName')).toEqual(false);
+})
+
+test('Auto generated primary key fields are not included in the Create<T>Mutation', () => {
+  const modelAST = `
+    """
+    @model
+    """
+    type User {
+      id: ID!
+      name: String!
+    }
```

Comment:
What happens for a model

```graphql
 """
    @model
    """
    type User {
      id: ID!
      name: String!
      _id: GraphbackObjectID
    }
```

?

Reply:
It would probably just not include both - but then that is the fault of the user and should not even be done. These are documented as auto-incrementing so the user should not do this.

Reply:
Actually upon checking, this is validated earlier in the execution:

`Error: UserFour type should not have two potential primary keys: "_id" and "id". Use '@id' annotations to indicate which one is to be used.`

Comes from: https://github.com/aerogear/graphback/blob/a84dcddade1b55b2955841ac2764eefec261f4f4/packages/graphback-core/src/db/getPrimaryKey.ts#L35

Reply:
Agree. I copy pasted in a rush and our primary key detection will throw an error the copied code above.  

I meant to have take the below business case. 

```graphql
    """
    @model
    """
    type User {
      """
     @id
     """
      key: String!
      name: String!
      _id: GraphbackObjectID
      id: ID
    }
```

Here the primary key is `key`. This means that `_id` and `id` will not be included right?

Reply:
Got you. That is an uncounted use case - fixing it now.

Reply:
Fixed now in https://github.com/aerogear/graphback/pull/2171/commits/d4916a25e8cf40d055b383834f60fee323037c2e